hoot-services/src/main/java/hoot/services/controllers/grail/MergeOsmFilesCommand.java

```diff
@@ -48,23 +49,31 @@
 
     private final Map<String, Object> substitutionMap = new HashMap<>();
     private final Class<?> caller;
-
-    private static final FilenameFilter filter = new FilenameFilter() {
-        @Override
-        public boolean accept(File f, String name) {
-            // We want to find only .osm files
-            return name.endsWith(".osm");
-        }
-    };
+    private final FilenameFilter filter;
 
     MergeOsmFilesCommand(String jobId, GrailParams params, String debugLevel, Class<?> caller) {
         super(jobId, params);
         logger.info("Params: " + params);
 
         this.caller = caller;
 
+        List<String> options = Collections.emptyList();
+        if (params.getAdvancedOptions() != null) {
+            options = params.getAdvancedOptions().entrySet().stream().map(ent -> {
+                return ent.getKey() + "=" + ent.getValue();
+            }).collect(Collectors.toList());
+        }
+
         substitutionMap.put("DEBUG_LEVEL", debugLevel);
+        substitutionMap.put("HOOT_OPTIONS", toHootOptions(options));
         substitutionMap.put("OUTPUT", params.getOutput());
+
+        filter = new FilenameFilter() {
+            @Override
+            public boolean accept(File f, String name) {
+                return name.matches(params.getInput1());
+            }
+        };
```

Comment:
not sure if this is too clever but this can be converted to:
```
filter = (File file, String name) -> name.matches(params.getInput1());
```

Reply:
Not sure if the anonymous function has to implement the FilenameFilter interface accept method.  If it works this way, you can push the change.